### PR TITLE
lock/rpc: change rpcPath to be called serviceEndpoint.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
lock/rpc: change rpcPath to be called serviceEndpoint.

## Motivation and Context
This is a cleanup to ensure proper naming.

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/lock-rpc-server.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"fmt"
 	"math/rand"
 	"net/rpc"
-	"path"
+	pathutil "path"
```
Comment (balamurugana) [inline cmd/lock-rpc-server.go]: why we need to alias `path` package into `pathutl`?

Comment (harshavardhana) [reply cmd/lock-rpc-server.go]: the problem is that `path` variable generally just conflicts with the package name and sadly Go doesn't complain properly .. it is just better as a safety. I can keep it `path` as well i don't forse usage if `path` as a variable anytime in future.

Comment (balamurugana) [reply cmd/lock-rpc-server.go]: I do not see any `path` variable currently.  Even in future, why someone will bring generic `path` variable name to conflict with `path` package?

Comment (harshavardhana) [reply cmd/lock-rpc-server.go]: It happens in posix.go but it is also naturally possible to use a local variable by someone . which in turn generates this problem. I can keep it `path` for now. 

One more place is namespace-lock.go `nsParam`

Comment (balamurugana) [reply cmd/lock-rpc-server.go]: a struct having `path` field is not a problem but a variable naming `path` will confuse in general.  That should be avoided.

Comment (harshavardhana) [reply cmd/lock-rpc-server.go]: Look at the usage of lock() , unlock() see the whole code.. @balamurugana

**Review by balamurugana — APPROVED** [review]

**Review by fwessels — APPROVED** [review]

**Pull request merged**